src/client/Microsoft.Identity.Client/AuthenticationResultMetadata.cs

```diff
@@ -22,5 +22,22 @@ public AuthenticationResultMetadata(TokenSource tokenSource)
         /// The source of the token in the result.
         /// </summary>
         public TokenSource TokenSource { get; }
+
+        /// <summary>
+        /// Total time (in ms) spent to service this request, in ms. Includes time spent making Http Requests <see cref="TimeSpentInHttp"/>, time spent
+        /// in token cache callbacks <see cref="TimeSpentInCache"/>, time spent in MSAL and context switching.
+        /// </summary>
+        public long TimeSpentInTotal { get; set; }
+
+        /// <summary>
+        /// Time (in ms) MSAL spent in reading and writing to the token cache, i.e. in the OnBeforeAccess, OnAfterAccess etc. callbacks. 
+        /// Does not include internal MSAL logic for searching through the cache once loaded.
```

Comment:
Should this be included to measure the whole cache operation? Otherwise, only measuring accessing serialization cache, the property name might be misleading then.